compiler/rustc_ty_utils/src/needs_drop.rs

```diff
@@ -165,7 +181,7 @@ where
                         if self.reveal_coroutine_witnesses {
```

Comment:
This is the flag we could set to true `list_significant_drop_tys` and I think eliminate the `_raw` function above, rigbht?

Reply:
Yes. I will also add a documentation to `list_significant_drop_tys` to explain why one should avoid running the query before `mir_promoted` is computed. It would lead to query cycle if one tries.